File: src/main/java/org/parham/seasonjob/commands/default_cmd/Leave.java
package org.parham.seasonjob.commands.default_cmd;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.parham.seasonjob.commands.SeasonCommand;
import org.parham.seasonjob.data.member.MemberManager;
import org.parham.seasonjob.data.sender.Messages;

import java.util.ArrayList;
import java.util.List;

public class Leave implements SeasonCommand {

    @Override
    public boolean hasPermission() {
        return true;
    }

    @Override
    public boolean isPlayerOnly() {
        return true;
    }

    @Override
    public String getPermission() {
        return "seasonjob.leave";
    }

    @Override
    public int getNeedArguments() {
        return 0;
    }

    @Override
    public String getUsage() {
        return "";
    }

    @Override
    public void execute(CommandSender sender, Command cmd, String label, String[] args, boolean isLeader, boolean hasAccess) {
        Player player = (Player) sender;

        if ((isLeader && !hasAccess) || (!isLeader && hasPermission() && !player.hasPermission(getPermission()))) {
            player.sendMessage(Messages.getMessage("deny-permission-leave"));
        } else if (!MemberManager.contains(player.getUniqueId())) {
            player.sendMessage(Messages.getMessage("unemployed-leave"));
        } else if (isLeader) {
            player.sendMessage(Messages.getMessage("leave-leader"));
        } else {
            MemberManager.delete(player.getUniqueId());
            player.sendMessage(Messages.getMessage("success-leave"));
        }
    }

    @Override
    public List<String> getCompletions(CommandSender sender, String[] args) {
        return new ArrayList<>();
    }
}

File: src/main/kotlin/ir/parham/seasonJob/Commands/Default/Leave.kt
package ir.parham.seasonJob.Commands.Default

import Libs.API.ir.parham.SeasonJobsAPI.Commands
import Libs.API.ir.parham.SeasonJobsAPI.Senders.Message
import ir.parham.SeasonJobsAPI.Actions.Member
import org.bukkit.Bukkit
import org.bukkit.OfflinePlayer
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.UUID
import kotlin.collections.List

class Leave : Commands {
    override fun runner(sender: CommandSender, args: Array<out String>) {
        val message = Message()
        if (sender !is Player) {
            // deny console
            sender.sendMessage(message.get(Bukkit.getOfflinePlayer(UUID.randomUUID()), "denyConsole"))
        } else if (!sender.hasPermission("seasonjobs.leave")) {
            // deny perm
            sender.sendMessage(message.get(sender, "denyPerm"))
        } else if (!Member().contains(sender.uniqueId)) {
            // player is employed
            sender.sendMessage(message.get(sender, "playerIsEmploy"))
        } else {
            remover(sender)
            sender.sendMessage(message.get(sender, "leaveSuccess"))
            // leaved
        }
    }

    override fun completer(sender: CommandSender, args: Array<out String>): List<String> {
        return listOf("")
    }

    override fun remover(target: OfflinePlayer) {
        Member().remove(target.uniqueId)
    }

    override fun adder(target: OfflinePlayer) {
        TODO("Not yet implemented")
    }

    override fun setter(target: OfflinePlayer) {
        TODO("Not yet implemented")
    }
}